contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/oracle/interfaces/StoreInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OracleAncillaryInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleV2Interface.sol";
+
+import "@uma/core/contracts/common/implementation/AncillaryData.sol";
+import "@uma/core/contracts/common/implementation/AddressWhitelist.sol";
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+
+contract OptimisticArbitrator is Testable {
```

Comment:
Is this meant to be an example contract that someone would build on from or is it meant to be an UMA-developed (and audited) intermediary contract that someone uses to add the delete-after-dispute functionality? If the latter, would it make more sense just to allow a user to opt-in to deleting the dispute afterwards in the normal OO? Modifying the existing OO has a few advantages -- one being overall simplicity, but the other being a large reduction in gas costs.

It might be useful to briefly explain the purpose in the comments of this contract and the description of the PR.

Reply:
this is meant as an example contract. I agree we should add a description. This contract is meant to show `Arbitration pattern where request/proposal/dispute is submitted all at the same time. OO is not really used, only the DVM for arbitration is.` the point is to show how to build the escalation game directly in your contract and then ONLY arbitrate bonds within the OO. you could extend your contract implementation with this, resulting in internal escalation game support.

Reply:
I've added the following description to the contract:
```
This contract enables assertions to be made to the Optimistic Oracle and assertions to be ratified
by the UMA Data Verification Mechanism. An assertion consists of using the requestPrice and proposePrice
methods of the OptimisticOracle to confirm or deny the response to a question. To ratify an assertion,
an initial assertion must be made and disputed, such that the question is scaled to the DVM, where UMA token
holders vote on it.
```